### PR TITLE
add event payload to flask.g

### PR DESCRIPTION
**Pull request by itajaja**
this allow cool things like having a logger use these properties

### CODE REVIEW

**Comment by taion** [comment]
see https://github.com/4Catalyzer/flask-resty/blob/bed083d4bb443d489124bf66d54ad4fba5047119/flask_resty/context.py

**Comment by itajaja** [comment]
that's fair. just read that on the documentation

**Comment by taion** [comment]
how do you plan on using this btw?

**Comment by itajaja** [comment]
```py
class PollerFormatter(logging.Formatter):
    def format(self, record):
        assert flask.has_app_context()
        record.topic_name = flask.g.payload["topic"]
        record.message_id = flask.g.payload["message"].get("id", "<NO ID>")

        return super().format(record)


def run():  # pragma: no cover
    set_logger_level(app.logger, logging.INFO)
    flask.logging.default_handler.setFormatter(
        PollerFormatter(
            "[%(asctime)s] %(levelname)s in %(module)s %(topic_name)s(%(message_id)s): %(message)s"
        )
    )
    poller.start()
```

minus switching `g` with context

**Comment by taion** [comment]
i don't suppose you'd care to make that available here?

**Comment by taion** [comment]
in this library as well, i mean

**Comment by itajaja** [comment]
I don't think so. If we had documentation, that's where that snippet might go

**Comment by taion** [comment]
i was thinking i'd be happy to re-use as-is, if it's a generic enough format

**Comment by itajaja** [comment]
that than could go in a shared 4C utils library, which is in our shared roadmap. wdyt?

**Comment by itajaja** [comment]
this specifically is not very generic:

```py
record.message_id = flask.g.payload["message"].get("id", "<NO ID>")
```

as other users might not have the convention of using the `id` field

File: tqp/flask.py [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
 
 # -----------------------------------------------------------------------------
 
+UNDEFINED = object()
+CTX_PAYLOAD_KEY = 'payload'
+
+# -----------------------------------------------------------------------------
+
+
+@property
```
Comment (taion) [inline tqp/flask.py]: uhh i don't think you can do this?

what?

Comment (itajaja) [reply tqp/flask.py]: lol yeah. I was writing a class and then refactored this. oops

File: tqp/flask.py [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
 
 # -----------------------------------------------------------------------------
 
+UNDEFINED = object()
+CTX_PAYLOAD_KEY = 'payload'
+
+# -----------------------------------------------------------------------------
+
+
+def _get_get_tqp_context():
```
Comment (taion) [inline tqp/flask.py]: ?

Comment (itajaja) [reply tqp/flask.py]: blub

**Pull request merged**